src/main/java/com/labsynch/labseer/api/ApiLsThingController.java

```diff
@@ -69,11 +69,11 @@
 	@RequestMapping(value = "/search", method = RequestMethod.GET)
 	@ResponseBody
 	public ResponseEntity<String> genericLsThingSearch(@RequestParam(value="lsType", required = false) String lsType, @RequestParam("q") String searchQuery,
-														@RequestParam(value = "with", required = false) String with) {
+														@RequestParam(value="lsKind", required = false) String lsKind, @RequestParam(value = "with", required = false) String with) {
 		HttpHeaders headers = new HttpHeaders();
 		headers.add("Content-Type", "application/json");
 		try {
-			Collection<LsThing> results = lsThingService.findLsThingsByGenericMetaDataSearch(searchQuery, lsType);
+			Collection<LsThing> results = lsThingService.findLsThingsByGenericMetaDataSearch(searchQuery, lsType, lsKind);
```

Comment:
If that lsKind parameter is optional, do we need to add any if/else logic? Or is lsKind just passed through as null and handled in LsThingServiceImpl?

Reply:
Yes the logic is handled further down in the the implementation.  I thought the same thing as you, but it was the pattern that was already their for lsType so I followed it.